pkg/util/resize/configspec.go

```diff
@@ -269,6 +270,40 @@ func compareExtraConfig(
 	outCS.ExtraConfig = util.MergeExtraConfig(ci.ExtraConfig, extraConfig)
 }
 
+// compareConsolePreferences compares the console preferences settings in the Config Spec.
+func compareConsolePreferences(
+	ci vimtypes.VirtualMachineConfigInfo,
+	cs vimtypes.VirtualMachineConfigSpec,
+	outCS *vimtypes.VirtualMachineConfigSpec) {
+
+	if cs.ConsolePreferences == nil {
+		return
+	}
+
+	if ci.ConsolePreferences == nil {
+		// If configInfo console preferences is nil, there is nothing to compare.
+		// set desired values
+		outCS.ConsolePreferences = &vimtypes.VirtualMachineConsolePreferences{
+			PowerOnWhenOpened:        cs.ConsolePreferences.PowerOnWhenOpened,
+			EnterFullScreenOnPowerOn: cs.ConsolePreferences.EnterFullScreenOnPowerOn,
+			CloseOnPowerOffOrSuspend: cs.ConsolePreferences.CloseOnPowerOffOrSuspend,
+		}
+
+		return
+	}
+
+	if ci.ConsolePreferences.PowerOnWhenOpened != cs.ConsolePreferences.PowerOnWhenOpened ||
```

Comment:
Is compare just the _pointers_ here what we want? I guess that's OK if we later see if after the `cmpPtr()`'s if fields`ConsolePreferences` are all nil and then just nil out `outCS.ConsolePreferences`?

Reply:
Agree. going with what you suggested.